setup.py

```diff
@@ -410,6 +464,20 @@ def get_hdf5_version(headername):
         # Don't delete this argument. It maybe useful for distutils
         # when adding more flags later on
         # sys.argv.remove(arg)
+    elif arg.find('--use-pkgconfig') == 0:
```

Comment:
this whole
```
args = sys.argv[:]
for arg in args:
     if arg.find('--hdf5=') == 0:
```
is quite ugly. Can't we just use `argparse` or even `getopt` from the standard library?